libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -398,6 +400,9 @@ private synchronized void enableLocationCollector(final boolean enable) {
       @Override
       public void run() {
         try {
+          // lazy install if required
+          LocationCollectionClient.install(applicationContext,
+            TimeUnit.HOURS.toMillis(DEFAULT_SESSION_ROTATION_INTERVAL_HOURS));
```

Comment:
Telemetry will be enabled during [map init process](https://github.com/mapbox/mapbox-gl-native-android/blob/238abb470a3774ca924805ee25467382d2e3fb3c/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/TelemetryImpl.java#L33), this will neutralize the delay initialize. IMHO, it will be better if we put the delay process in `LocationCollectionClient.install`